### DISCUSSION TITLE
Is kuiper suitable to do the job of  platform protocol transform?

### DISCUSSION THREAD

**Discussion by wallace-hu** [lf-edge/ekuiper#1005] [Q&A]
I am using the edgex framework to connect to clound platform,  As you know different cloud platform has different message format.  So in my design, I use kuiper to do the main job of transforming the edgex events message(one kind of Json fromat message) to the platform required message format  by using the kuiper "dataTemplate" feature.  

I wonder that in this scenario if the data load increase, Can the kuiper bear the load?   Can kuiper be used in this way when you design it?

**Comment by ngjaying** [comment]
This is one of the common scenario that eKuiper designed for.

**Reply by jinfahua** [reply]
The data template uses Golang native feature, it should not have too much performance overhead. I think it should be no problem.
